File: Solutions/1457. Pseudo-Palindromic Paths in a Binary Tree/1457. Pseudo-Palindromic Paths in a Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    cnt=[0]*10
    def pseudoPalindromicPaths (self,root,odds=0,res=0) -> int:
        res=0
        if root:
            self.cnt[root.val]+=1
            odds+=1 if self.cnt[root.val]%2==1 else -1
            if root.left==root.right:
                res=1 if odds<2 else 0
            else:
                res=self.pseudoPalindromicPaths(root.left,odds)+self.pseudoPalindromicPaths(root.right,odds)
            self.cnt[root.val]-=1
            odds+=1 if self.cnt[root.val]%2==-1 else 1
        return res

File: Solutions/1457. Pseudo-Palindromic Paths in a Binary Tree/1457. Pseudo-Palindromic Paths in a Binary Tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func pseudoPalindromicPaths (root *TreeNode) int {
    cnt:=make([]int,10)
    return dfs(root,&cnt,0)
}
func dfs(root *TreeNode,cnt *[]int,odds int) int{
    res:=0
    if root!=nil{
        (*cnt)[root.Val]+=1
        if (*cnt)[root.Val]%2==1{
            odds+=1
        }else{
            odds-=1
        }
        if root.Left==nil &&root.Right==nil{
            if odds<2{
                res=1
            }else{
                res=0
            }
        }else{
            res=dfs(root.Left,cnt,odds) + dfs(root.Right,cnt,odds) 
        }
        (*cnt)[root.Val]-=1
        if (*cnt)[root.Val]%2==1{
            odds-=1
        }else{
            odds+=1
        }
    }
    return res
}